src/games/strategy/thread/LockUtil.java

```diff
@@ -21,42 +24,54 @@
  * you are considering your ambitious multi-threaded code a mistake, and you are trying to limit the damage.
  * <p>
  */
-public class LockUtil {
+public final class LockUtil {
+  private static final LockUtil instance = new LockUtil();
```

Comment:
Prefer to use enum singleton's. We should have a number of examples, but basically make this class an enum with a single 'singleton' instance.